### PR TITLE
feat: Various improvements

### PR DESCRIPTION
**Pull request by raphael0202**
- improve bot crawling list
- add include/exclude parameter (Solves #38)
- switch to Python3.11 and more recent dependencies

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -109,12 +106,23 @@ async def knowledge_panel(
...
 ):
     """
     FacetName is the model that have list of values
     facet_tag are the list of values connecting to FacetName
     eg:- category/beer, here beer is the value
     """
+    if include is not None and exclude is not None:
+        raise HTTPException(status_code=400, detail="include and exclude parameters are exclusive")
+
+    panel_names = {PanelName.hunger_game, PanelName.wikidata}
```
Comment (alexgarel) [inline app/main.py:120]: Those are the defaults ?

In this case we should document it.

I was thinking panel_names would have them all, and we would change off-server code to only ask the one we want. 

But it's ok for a transition period.

**Review by alexgarel — APPROVED** [review]
Great !

Only thing, I would prefer if you can keep tests, using include parameter if needed.

File: app/main.py (hunk 1 above)
Comment (raphael0202) [reply app/main.py:120]: As other panels are expensive for the server, I think it's better to have sane default. I'm adding it to the documentation.

File: tests/test_main.py [hunk 2]
```diff
@@ -70,9 +70,7 @@ def test_knowledge_panel_no_value(client, monkeypatch):
     response = client.get("/knowledge_panel?facet_tag=origin")
     assert response.status_code == 200
     result = response.json()
-    assert set(result["knowledge_panels"].keys()) == {"Quality", "LastEdits"}
-    assert len(result["knowledge_panels"]["Quality"]["elements"]) == 1
-    assert len(result["knowledge_panels"]["LastEdits"]["elements"]) == 1
+    assert len(result["knowledge_panels"]) == 0
```
[the remark this answers is not in the document]
Comment (raphael0202) [reply tests/test_main.py:73]: yes I'm changing it!

**Pull request merged**